ApplicationInsights/JavaScriptSDK/Telemetry/PageViewManager.ts

```diff
@@ -54,50 +56,52 @@ export class PageViewManager {
         }
 
-        var pageViewSent = false;
-        var customDuration = undefined;
 
-        if (PageViewPerformance.isPerformanceTimingSupported()) {
-            var start = PageViewPerformance.getPerformanceTiming().navigationStart;
-            customDuration = PageViewPerformance.getDuration(start, +new Date);
-
-            if (!PageViewPerformance.shouldCollectDuration(customDuration)) {
-                customDuration = undefined;
-            }
-        } else {
+        // if performance timing is not supported by the browser, send the page view telemetry with the duration provided by the user. If the user
+        // do not provide the duration, set duration to undefined
+        if (!PageViewPerformance.isPerformanceTimingSupported()) {
             this.appInsights.sendPageViewInternal(
                 name,
                 url,
                 !isNaN(duration) ? duration : undefined,
                 properties,
                 measurements);
             this._channel.flush();
-            pageViewSent = true;
+
+            // no navigation timing (IE 8, iOS Safari 8.4, Opera Mini 8 - see http://caniuse.com/#feat=nav-timing)
+            _InternalLogging.throwInternal(
+                LoggingSeverity.WARNING,
+                _InternalMessageId.NavigationTimingNotSupported,
+                "trackPageView: navigation timing API used for calculation of page duration is not supported in this browser. This page view will be collected without duration and timing info.");
+
+            return;
         }
 
-        if (!pageViewSent && (this.overridePageViewDuration || !isNaN(duration))) {
-            // 1, 2, 4 cases
+        var pageViewSent = false;
+        var customDuration = undefined;
+
+        // if the performance timing is supported by the browser, calculate the custom duration
+        var start = PageViewPerformance.getPerformanceTiming().navigationStart;
+        customDuration = PageViewPerformance.getDuration(start, +new Date);
+        if (!PageViewPerformance.shouldCollectDuration(customDuration)) {
+            customDuration = undefined;
+        }
+
+        // if the user has provided duration, send a page view telemetry with the provided duration. Otherwise, if
+        // overridePageViewDuration is set to true, send a page view telemetry with the custom duration calculated earlier
+        if (this.overridePageViewDuration || !isNaN(duration)) {
             this.appInsights.sendPageViewInternal(
                 name,
                 url,
                 !isNaN(duration) ? duration : customDuration,
                 properties,
                 measurements);
-                this._channel.flush();
+            this._channel.flush();
             pageViewSent = true;
         }
 
+        // now try to send the page view performance telemetry
         var maxDurationLimit = 60000;
-
-        if (!PageViewPerformance.isPerformanceTimingSupported()) {
```

Comment:
you'll be adding this back again right when the return false on isPerformanceTimingSupported switches to looking at browser functionality? If so, can you track this both through a task/

Reply:
I did not remove it. I moved it up to line 62. With this pull request, I did not change any functionality to what it was. Just cleaning this method as it was confusing to me. Now it's a little bit more readable.

Reply:
Once you're free, we can pair on this to discuss the change as it is easier to walk through it face to face.